### PR TITLE
Update sequence diagrams

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

**Comment by Nil20** [comment]
@Zangetsu101 We made some change into the search service's work. So I guess we should update the operations of `upsertEvent` here as well, right?

**Comment by Zangetsu101** [comment]
> @Zangetsu101 We made some change into the search service's work. So I guess we should update the operations of `upsertEvent` here as well, right?

yes please

File: sequence-diagrams/backend/create-birth-registration-field-agent.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    participant Country-Config
+
+    GraphQL gateway->>Workflow: POST /create-record
+    Workflow--)Hearth: Fetch task from identifier(draft-id)
+    Note over Workflow,Hearth: Check existing draft-IDs. If found, return previous declaration info to gateway.
+
+    Workflow--)Search: Search for duplicates
+
+    Workflow--)Documents: POST certificate details to /upload
+    Documents->>Minio: Upload certificate documents
```
Comment (Zangetsu101) [inline sequence-diagrams/backend/create-birth-registration-field-agent.md:31]:
```suggestion
    Workflow--)Documents: POST attachment details to /upload
    Documents->>Minio: Upload attachment documents
```

File: sequence-diagrams/backend/create-birth-registration-field-agent.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    Workflow->>Hearth: Save bundle
+    Note over Workflow,Hearth: Get hearth response for all entries
+
+    Note over Workflow: Merge changed resources<br /> into record with <br /> hearth's response bundle
+
+    Workflow--)Metrics: POST bundle to /events/{event}/sent-notification
+    Metrics->>Influx DB: Write user audit point "IN_PROGRESS"
+
+    loop location levels 4, 3, 2
+        Metrics->>Workflow: Get parent of Location
```
Comment (Zangetsu101) [inline sequence-diagrams/backend/create-birth-registration-field-agent.md:52]:
```suggestion
        Metrics->>Hearth: Get parent of Location
```

File: sequence-diagrams/backend/create-birth-registration-field-agent.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        Workflow->>Hearth: Get parent of Location
+    end
+    Note over Metrics,Hearth: Generate time logged point
+
+    loop location levels 4, 3, 2
+        Metrics->>Workflow: Get parent of Location
```
Comment (Zangetsu101) [inline sequence-diagrams/backend/create-birth-registration-field-agent.md:58]:
```suggestion
        Metrics->>Hearth: Get parent of Location
```

File: sequence-diagrams/backend/create-birth-registration-field-agent.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    Note over Search,Hearth: Compose new history entry
+
+    Search->>ElasticSearch: Index composition
+    Workflow->>Hearth: Save bundle to Hearth
+    Note over Workflow,Search: For duplicate declarations, ends here
+
+    Workflow--)Config: Check if notification is enabled
+    Workflow--)Search: Index bundle
+
+    Workflow--)Notifications: Send notification if enabled
```
Comment (Zangetsu101) [inline sequence-diagrams/backend/create-birth-registration-field-agent.md:91]:
```suggestion
    Workflow--)Config: Check if notification is enabled
    Workflow--)Notifications: Send notification if enabled

    Workflow--)Search: Index bundle
```

Comment (Zangetsu101) [reply sequence-diagrams/backend/create-birth-registration-field-agent.md:91]: We should also mention how search services fetches data from hearth for populating location details

File: sequence-diagrams/backend/create-birth-registration-field-agent.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    Notifications->>Country-Config: POST /notification
+    Country-Config->>Config: GET Application Config from /publicConfig
+    Config->>Country-Config: Get config from /application-config
+    Note over Workflow,Notifications: Get APPLICATION NAME, Notification Delivery Methods
```
Comment (Zangetsu101) [inline sequence-diagrams/backend/create-birth-registration-field-agent.md:95]:
```suggestion
    Note over Config,Notifications: Get APPLICATION NAME, Notification Delivery Methods
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: sequence-diagrams/backend/create-birth-registration-field-agent.md (hunk 4 above)
Comment (Nil20) [reply sequence-diagrams/backend/create-birth-registration-field-agent.md:91]: The order of the flow is currently how it is demonstrated here. We could update the flow to serialize the notification flow.

File: sequence-diagrams/backend/mark-birth-as-downloaded.md
Comment (Zangetsu101) [inline sequence-diagrams/backend/mark-birth-as-downloaded.md:1]: This still seems to have OpenHIM as a participant

File: sequence-diagrams/backend/mark-birth-as-reinstated.md
Comment (Zangetsu101) [inline sequence-diagrams/backend/mark-birth-as-reinstated.md:1]: missing the metrics call

**Review by Zangetsu101 — COMMENTED** [review]
@Nil20 There were a couple of calls added to the webhooks microservice which we will need to include in the register flow

File: sequence-diagrams/backend/create-birth-registration-field-agent.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        Workflow->>Hearth: Get parent of Location
+    end
+    Note over Metrics,Hearth: Generate time logged point
+
+    loop location levels 4, 3, 2
+        Metrics->>Hearth: Get parent of Location
+        Workflow->>Hearth: Get parent of Location
+    end
+    Note over Metrics,Hearth: Generate declaration started point
+    Metrics->>Influx DB: Write audit points
```
[the remark this answers is not in the document]
Comment (Nil20) [reply sequence-diagrams/backend/create-birth-registration-field-agent.md:62]: Yeah. This probably came in when I was find-and-replace-ing OPENHIM with workflow. Should be rather this.
```
    loop location levels 4, 3, 2
        Metrics->>Hearth: Get parent of Location
    end
```

File: sequence-diagrams/backend/create-birth-registration-registrar.md [hunk 7]
```diff
@@ -22,281 +22,96 @@ sequenceDiagram
...
+    Note over Workflow,Hearth: Get hearth response for all entries
+
+    Note over Workflow,Hearth: To registered state
+
+    Note over Workflow: Merge changed resources<br /> into record with <br /> hearth's response bundle
+
+    Workflow--)Search: Index bundle
+    Workflow--)Metrics: POST bundle to /events/{event}/registered
+
+    Workflow--)Search: Index bundle
```
[the remark this answers is not in the document]
Comment (Nil20) [reply sequence-diagrams/backend/create-birth-registration-registrar.md:104]: Actually the second is for indexing the rejected bundle. Would it be necessary to include that in this diagram?

File: sequence-diagrams/backend/mark-birth-as-archived.md
[the remark this answers is not in the document]
Comment (Nil20) [reply sequence-diagrams/backend/mark-birth-as-archived.md:1]: There are no call to metrics from the archive endpoint. We actually missed it to add it in the endpoint.

File: sequence-diagrams/backend/mark-birth-as-assigned.md
[the remark this answers is not in the document]
Comment (Nil20) [reply sequence-diagrams/backend/mark-birth-as-assigned.md:1]: Yeah, also missed here. Also the call is also missing in the unassign handler as well.

File: sequence-diagrams/backend/mark-birth-as-certified.md [hunk 8]
```diff
@@ -13,54 +12,63 @@ sequenceDiagram
...
         Workflow->>Hearth: Get parent of Location
     end
     Note over Metrics,Hearth: Generate certification point
 
     loop location levels 4, 3, 2
-        Metrics->>OpenHIM: Get parent of Location
-        OpenHIM->>Workflow: Get parent of Location
+        Metrics->>Hearth: Get parent of Location
         Workflow->>Hearth: Get parent of Location
     end
```
[the remark this answers is not in the document]
Comment (Nil20) [reply sequence-diagrams/backend/mark-birth-as-certified.md:73]: Yeah like the one you mentioned on a previous one.

File: sequence-diagrams/backend/create-birth-registration-registrar.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Nil20) [reply sequence-diagrams/backend/create-birth-registration-registrar.md:102]: Removing this for "rejected"

File: sequence-diagrams/backend/mark-birth-as-reinstated.md [hunk 9]
```diff
@@ -57,4 +57,23 @@ sequenceDiagram
...
+    end
+    Note over Metrics,Hearth: Generate certification point
+
+    loop location levels 4, 3, 2
+        Metrics->>Hearth: Get parent of Location
+    end
+    Note over Metrics,Hearth: Generate payment point
+
+    Metrics->>Hearth: Get previous task of Task History
+    Note over Metrics,Hearth: Generate event duration point
```
Comment (Zangetsu101) [inline sequence-diagrams/backend/mark-birth-as-reinstated.md:76]: These seem a bit out of place here

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**